app/components/redirect.tsx

```diff
@@ -0,0 +1,27 @@
+import { useNavigate } from '@remix-run/react';
+import type { To } from '@remix-run/router';
+import type { PropsWithChildren } from 'react';
+import type { NavigateOptions } from 'react-router/dist/lib/context';
```

Comment:
is this type available in remix's router?

Reply:
only in react-router unfortunately so I changed it to `import type { NavigateOptions } from 'react-router'`